applications/ContactStructuralMechanicsApplication/custom_processes/master_slave_process.cpp

```diff
@@ -19,88 +19,104 @@
 
 namespace Kratos
 {
-    void MasterSlaveProcess::Execute()
+void MasterSlaveProcess::Execute()
+{
+    KRATOS_TRY;
+    
+    ModelPart& contact_model_part = mrThisModelPart.GetSubModelPart("Contact");
+    
+    // Now we iterate over the nodes
+    NodesArrayType& nodes_array = mrThisModelPart.Nodes();
+    const int num_nodes = static_cast<int>(nodes_array.size());
+    
+    // Now we iterate over the conditions
+    ConditionsArrayType& conditions_array = mrThisModelPart.Conditions();
+    const int num_conditions = static_cast<int>(conditions_array.size());
+    
+    // Creating a buffer for parallel vector fill
+#ifdef _OPENMP
+    const int num_threads = OpenMPUtils::GetNumThreads();
+#else
+    const int num_threads = 1;
+#endif
+    std::vector<std::size_t> index_node;
+    std::vector<std::vector<std::size_t>> index_node_buffer(num_threads);
+    std::vector<std::size_t> index_cond;
+    std::vector<std::vector<std::size_t>> index_cond_buffer(num_threads);
+    
+#ifdef _OPENMP
```

Comment:
you don't need this. Actually the OpenMPutils is designed exactly for things like "ThisThread" to work even if openmp is not defined